appendix/matching/main.tex

```diff
@@ -0,0 +1,503 @@
+\chapter{An introduction to matching games}
+\label{app:matching}
+
+\graphicspath{{appendix/matching/paper/img/}}
+
+Matching games form a part of game theory that were formally introduced by Gale
+and Shapley in their seminal work~\cite{Gale1962}. These games allow for the
+allocation of resources and partnerships in a mathematically fair way.
+Typically, a matching game is defined by two sets of players (called parties)
+that each have preferences over at least some of the elements of the other set.
+The objective of the game is then to find a mapping between the sets of players
+in which everyone is \emph{happy enough} with their match(es).
+
+This appendix does not contain any novel mathematics, but it does offer an
+introduction to matching games and their variants. Studying this branch of
+mathematics has contributed to a significant amount of the research conducted
+for this thesis, hence its inclusion here. That research has culminated in the
+development of a Python library for solving various matching games, \matching.
+Among other uses, the \matching\ library proves instrumental in the practical
+implementation of the novel method described in Chapter~\ref{chp:kmodes}.
+
+The \matching\ library has been developed as a research tool and adheres to the
+best practices discussed in Chapter~\ref{chp:intro}. The current version of
+Matching has also been archived on Zenodo under~\doi{10.5281/zenodo.3931026}.
+Along with the source code being modularised and fully tested (using example,
+integration and property-based unit tests) with 100\% coverage, the library is
+documented extensively. Like the \edo\ library developed for the work in
+Chapter~\ref{chp:edo}, the \matching\ documentation is hosted online at
+\href{https://matching.readthedocs.io}{\nolinkurl{matching.readthedocs.io}}.
+The documentation has been written to maximise its effect as a resource for
+learning about matching games as well as for the software itself. Furthermore,
+the library is registered on the Python Package Index and is \pip-installable.
+
+\begin{listing}
+\begin{usagesh}
+> pip install matching
+\end{usagesh}
+\caption{Installing the \matching\ library via \pip}
+\end{listing}
+
+Matching games have applications in many fields where relationships between
+rational agents must be managed. Some example applications include: being able
+to inform on healthcare finance policy~\cite{Agarwal2017}; helping to reduce the
+complexity of automated wireless communication networks~\cite{Bayat2016}; and
+education infrastructure~\cite{Chiarandini2019}. Thus, having access to software
+implementations of algorithms that are able to solve such games is essential.
+
+The only current adversary to \matching\ is \matchingr~\cite{Tilly2018}.
+\matchingr\ is a package written in C++ with an R interface and its content
+overlaps well with that of \matching. However, the lack of a Python interface
+makes it less relevant to researchers and other users as Python's popularity
```

Comment:
Can we think of a way to reword this? I know what you're saying but a lot of R people would say that a Python library is irrelevant...

Reply:
There definitely is a better way of writing this; I lifted this paragraph directly from the JOSE paper on the matching library and I wasn't the biggest fan then